### PR TITLE
ORC-199. Add convert from CSV.

### PR DESCRIPTION
**Pull request by omalley**
This patch incorporates Carter's converter into the tool jar. I haven't made any tests yet.

### CODE REVIEW

**Comment by spasam** [comment]
@omalley Am interested in testing this with PostgreSQL dump output (CSV and TEXT). Is something blocking you from merging this? Thanks

**Comment by cartershanklin** [comment]
I think Owen mentioned he had incorporated the code into another tool, if so we should close the ticket with the right pointer.

The original project had customizable null strings and strict mode that were put there with Postgres in mind. Interested to hear your experiences when you try it outl

File: java/tools/src/java/org/apache/orc/tools/convert/ConvertTool.java [hunk 1]
```diff
@@ -84,7 +225,22 @@ static CommandLine parseOptions(String[] args) throws ParseException {
     options.addOption(
         Option.builder("o").longOpt("output").desc("Output filename")
             .hasArg().build());
-    CommandLine cli = new GnuParser().parse(options, args);
+    options.addOption(
```
Comment (spasam) [inline java/tools/src/java/org/apache/orc/tools/convert/ConvertTool.java]: Could you document the default values for these options in the help text?

File: java/tools/src/java/org/apache/orc/tools/convert/ConvertTool.java [hunk 2]
```diff
@@ -18,53 +18,178 @@
...
+        schemaFinder.addFile(file.getReader(file.filesystem.open(file.path)));
+      } else if (file.format == Format.ORC) {
+        System.err.println("Merging schema from " + file.path);
+        Reader reader = OrcFile.createReader(file.path,
+            OrcFile.readerOptions(conf)
+                .filesystem(file.filesystem));
+        schemaFinder.addSchema(reader.getSchema());
+      }
     }
     return schemaFinder.getSchema();
```
Comment (spasam) [inline java/tools/src/java/org/apache/orc/tools/convert/ConvertTool.java]: This is throwing NPE if no command line arguments are specified except for CSV file:

```
Exception in thread "main" java.lang.NullPointerException
	at org.apache.orc.tools.json.JsonSchemaFinder.getSchema(JsonSchemaFinder.java:321)
	at org.apache.orc.tools.convert.ConvertTool.buildSchema(ConvertTool.java:75)

```

**Comment by spasam** [comment]
Finally was able to test this end to end. I had to tweak CsvReader. Added customer converters for boolean and timestamp. I can upload changes in a separate pull request.

Ideally, if **json-schema** Driver command line option is changed to **schema** and CSV schema is determined on the fly (assuming there is a header line with column names), this would be awesome.

**Pull request merged**